### DISCUSSION TITLE
Album page preview

### DISCUSSION THREAD

**Discussion by mattl** [foocorp/hacienda#89] [Future enhancements]
Very early album page preview.

Contrast https://hacienda-prototype.glitch.me/album.html with https://libre.fm/artist/Joy+Division/album/Unknown+Pleasures+%28Collector%27s+Edition%29

All feedback is welcome. Dark mode and light mode CSS is provided. 

EDIT: https://hacienda-prototype-static.glitch.me/album-20240203.html is now a more current version of album.html linked above

**Comment by grdryn** [comment]
Nice work Matt...it's exciting to see progress here!

- I like the ordered track numbering compared to the alphabetized tracks on the libre.fm example.
- The fact that there's only a single entry for each track compared to the N variations on the live site is another obvious difference (I know the Collector's Edition has more tracks, but the Libre.fm page has more versions again). Do you expect that to be how it remains? Maybe different variations could be linked to the album track title or something?
- I presume that the tracks will eventually have their own page with more information?
- I really like how it looks on a mobile device. I think the font sizes could be reduced (at least on those smaller screens) though: the album title is only two words and would look better on a single line IMHO.
- Do you think that any other information on the album page would be useful? For example, optional external links, like Wikipedia or where to buy on Bandcamp or the artist's website, or maybe things like release date (date metadata could help with ordering releases chronologically on an artist page, if that's desirable)?

I look forward to seeing more, and eventually get to use this! :dog:

**Comment by mattl** [comment]
Congrats on being the first person to give feedback @grdryn -- much appreciated.

- I'd like albums on Libre.fm going forward to reflect the real album as released, and so we'd have listings for all the versions of an album. I won't pretend to have some magic to work out which version of an album is being listened to other than the metadata submitted. MusicBrainz data can be harnessed here, as well as a link to the item on Discogs or something similar.

- Track information, yes. Perhaps a track would show all the albums that track is on that we know of? 

- Mobile support is going to be much better on the new site. Most of my casual web browsing and 90% of my music listening is done on my phone now. That wasn't true in 2009.

- There are some freely licensed resources that describe an album, a track, an artist so it would be good to bring those all in too. Libre.fm can be a place to find out more about music. 

Look at all the information Wikidata has for example: https://www.wikidata.org/wiki/Q735573 -- it would be fun to figure out some ways to show that information.

**Comment by janerationx** [comment]
I like it. Big, clean and modern, although the fonts should be smaller for mobile, as @grdryn said.

**Comment by mattl** [comment]
I’ll look at getting a mobile improved version of the page up this weekend. I agree it’s not looking great right now.

**Comment by mattl** [comment]
Okay, a new preview of the album page slightly improved is now at https://hacienda-prototype-static.glitch.me/album-20240203.html

**Comment by hstriepe2** [comment]
Since i, eclectic listener, am new to this project and to libre, take my words with a grain of salt.

I like the preview album page, especially the release year is very important (to me, but also to be able to distinguish album versions).

Did you think of how to handle multiple artists already? We had several approaches from grooveshark to last to spot etc during the past, and probably being able to „click each artist“ is better than VA or just mentioning the main artist?

Discogs sure is a good source, but also very limited (25-60 calls/min): https://www.discogs.com/developers#page:home,header:home-rate-limiting

I‘m really interested in what and how new libre is going to be.

**Comment by ea-st** [comment]
Looking good so far, I especially like the mobile version of the site.
If I were to make any suggestions I think a lot of elements could be visually condensed similar to the image below.

![Librefm](https://github.com/foocorp/hacienda/assets/119245891/cd5f03c8-6f2f-4baa-ad5c-5f62d063b8d3)

While both look great I'm still split between if I prefer this new look or the centered list Libre.fm has currently.

Looking forward to seeing more!

**Comment by Vestal-1** [comment]
Hello Matt, thank you for your excelent work! I have these suggestions:
- I think length of tracks could be added as well, as in `1. Disorder (3:32)`, maybe spaced a little so that all the timestamps are in same column.
- Having a display for **Album Artist** would be a nice idea, since sometimes (very often) classical releases have the same title names, but different performers. See, for instance, Schoenberg: [Von Vintscher's "Piano Music"](https://www.discogs.com/master/1475119-Schoenberg-J%C3%BCrg-Von-Vintschger-Piano-Music) / [Paul Jacob's "Piano Music"](https://www.discogs.com/release/2491564-Arnold-Schoenberg-Paul-Jacobs-Piano-Music) [There are some cases where the title is the same, and there's multiple performers + different tracks in both albums (e.g. [1](https://musicbrainz.org/release/88b23d1e-708f-4c15-9e2f-7e6be23ec01d) / [2](https://musicbrainz.org/release/21fdabcb-215a-4d38-b5c6-8f5be37c3c25)), similar to @hstriepe2 comment, but I wouldn't know how to solve that problem].
